ddsreplayer/test/blackbox/mcap/McapFileReadTest.cpp

```diff
@@ -0,0 +1,199 @@
+using namespace eprosima::fastrtps::rtps;
+using namespace eprosima::fastrtps;
+using namespace eprosima::ddsrecorder::replayer;
+
+namespace test {
+
+// Publisher
+
+const unsigned int DOMAIN = 0;
+
+std::string topic_name = "/dds/topic";
+std::string data_type_name = "HelloWorld";
+
+std::string send_message = "Hello World";
+unsigned int index = 6;
+unsigned int downsampling = 3;
+
+} // test
+
+enum class DataTypeKind
+{
+    HELLO_WORLD,
+};
+
+void create_subscriber_replayer(
+        DataToCheck& data)
+{
+    // Create Subscriber
+    HelloWorldSubscriber subscriber(
+        test::topic_name,
+        static_cast<uint32_t>(test::DOMAIN),
+        11,
+        data);
+
+    std::cout << "subscriber created !!!!" << std::endl;
+
+    // Create a multiple event handler that handles all events that make subscriber and replayer stop
+    auto close_handler_replayer = std::make_shared<eprosima::utils::event::MultipleEventHandler>();
+    auto close_handler_subscriber = std::make_shared<eprosima::utils::event::MultipleEventHandler>();
+
+    // Run Participant
+    std::thread run_subscriber([&]
+            {
+                try
+                {
+                    subscriber.run();
+                }
+                catch (const eprosima::utils::InconsistencyException& e)
+                {
+                    logError(DDSREPLAYER_ERROR,
+                    "Error running subscriber. Error message:\n " <<
+                        e.what());
+                }
+                close_handler_subscriber->simulate_event_occurred();
+            });
+
+
+    // Configuration File path
+    std::string file_path = "resources/config_file.yaml";
+    eprosima::ddsrecorder::yaml::ReplayerConfiguration configuration(file_path);
+
+    // Create replayer instance
+    std::string input_file = "resources/helloworld_file.mcap";
+    auto replayer = std::make_unique<DdsReplayer>(configuration, input_file);
+
+    std::cout << "replayer created !!!!" << std::endl;
+
+    // Start replaying data
+    bool read_success;
+    std::thread process_mcap_thread([&]
+            {
+                try
+                {
+                    replayer->process_mcap();
+                    read_success = true;
+                }
+                catch (const eprosima::utils::InconsistencyException& e)
+                {
+                    logError(DDSREPLAYER_ERROR,
+                    "Error processing MCAP file. Error message:\n " <<
+                        e.what());
+                    read_success = false;
+                }
+                close_handler_replayer->simulate_event_occurred();
+            });
+
+    // Wait until signal arrives (or all messages in MCAP file sent)
+    close_handler_replayer->wait_for_event();
+
+    // Wait until signal arrives
+    close_handler_subscriber->wait_for_event();
+
+    // Disable inner pipe, which would abort replaying messages in case execution stopped by signal
+    replayer->stop();
```

Comment:
I believe there is no need to run this in a dedicated thread + using a signal handler. You can process the mcap in the main thread, and the execution will end when all messages are replayed.
The signal handler is used in the replayer tool to abort replaying messages when a user wishes to (via signal), but here I don't think we will be testing that.